crates/apollo-compiler/src/validation/validation_db.rs

```diff
@@ -434,10 +524,14 @@ pub fn validate_type_system(db: &dyn ValidationDatabase) -> Vec<ApolloDiagnostic
 
     diagnostics.extend(db.validate_extensions());
 
+    diagnostics.sort_by_key(location_sort_key);
     diagnostics
 }
 
-pub fn validate_executable(db: &dyn ValidationDatabase, file_id: FileId) -> Vec<ApolloDiagnostic> {
+pub fn validate_standalone_executable(
```

Comment:
I'd also have a wrapping query that iterates over all the executable files and calls this function. That way all executable files can be validated in one go and the user doesn't have to keep all file ids around. I imagine @EverlastingBugstopper  got a couple dozen of files they are trying to validate.

Reply:
^ Yeah this is a nice to have - was able to implement what I need w/o this but a nicer fn signature wouldn't require passing a file ID.